src/core/nativecall.c

```diff
@@ -299,6 +299,12 @@ char * MVM_nativecall_unmarshal_string(MVMThreadContext *tc, MVMObject *value, M
                 str = MVM_string_utf8_encode_C_string(tc, value_str);
         }
 
+#ifdef MVM_USE_MIMALLOC
+        char *libc_cstr = malloc(strlen(str) + 1);
+        strcpy(libc_cstr, str);
+        MVM_free(str);
```

Comment:
Better to store the result of `strlen` and then use `memcpy` instead of `strcpy`